packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -184,9 +184,27 @@ class CoverageCollector extends TestWatcher {
   }
 }
 
-Future<Map<String, dynamic>> collect(Uri serviceUri, bool Function(String) libraryPredicate) async {
+Future<Map<String, dynamic>> collect(Uri serviceUri, bool Function(String) libraryPredicate, { bool waitPaused = false }) async {
+  const int _kPausePollLimit = 5;
   final VMService vmService = await VMService.connect(serviceUri, compression: CompressionOptions.compressionOff);
   await vmService.getVM();
+  if (waitPaused) {
+    int i = 0;
+    Duration waitDuration = const Duration(seconds: 0);
+    while (i < _kPausePollLimit) {
```

Comment:
Can you simplify this at all with the `vmService.onIsolateEvent` stream instead of polling?

Reply:
This API proved challenging to use - it gives events for all isolate/ Future based. I adjusted the polling logic to be correct and more resilient instead.